lib/xdparser2.ts

```diff
@@ -642,23 +643,26 @@ function parseStyleCSSLike(str: string, xd: string, errorReporter: (msg: string,
 
 /**
  * Given an answer that might contain splits, and a split character, return
- * an array of all the split locations (where the split is to the right of the array value),
- * as well as the answer without the splits..
+ * an array of all the split locations
+ *
+ * Split location/index starts right after the first letter/character
+ * Example:
+ *  answerWithSplits: "abc"
+ *  The first split index would be between "a" and "b".
+ *  The index there would be 0
  *
  * @param answerWithSplits unparsed answer string
  * @param splitCharacter character to split on
- * @returns an array of split locations, and the answer without splits and with rebus replacements
+ * @returns an array of split locations
  */
 function parseSplitsFromAnswer(answerWithSplits: string, splitCharacter?: string): number[] | undefined {
   if (!splitCharacter) return undefined
   const splits = []
-  let answer = ""
-  for (var i = 0; i < answerWithSplits.length; i++) {
-    if (answerWithSplits.charAt(i) === splitCharacter) {
-      splits.push(i - 1 - splits.length)
-      continue
+  const characters = [...answerWithSplits] // account for unicode characters like emojis that could take up more than one utf-16 unit
```

Comment:
Note that this will not cover all emojis – especially modern ones that have race, and gender, etc. You have to do something more complicated to really get an accurate count if there are emojis in the string (see https://github.com/orling/grapheme-splitter or https://github.com/mathiasbynens/emoji-regex for more robust options).

Since this is an improvement over what we had before, it is ok to merge, but we should at least file an issue about this because it will come up again.